[2024-03-08T06:51:18.224+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: AIRLINES_ETL_PIPELINE.runSnowparkCode manual__2024-03-08T06:50:59.288974+00:00 [queued]>
[2024-03-08T06:51:18.240+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: AIRLINES_ETL_PIPELINE.runSnowparkCode manual__2024-03-08T06:50:59.288974+00:00 [queued]>
[2024-03-08T06:51:18.240+0000] {taskinstance.py:2193} INFO - Starting attempt 1 of 1
[2024-03-08T06:51:18.297+0000] {taskinstance.py:2214} INFO - Executing <Task(SnowflakeOperator): runSnowparkCode> on 2024-03-08 06:50:59.288974+00:00
[2024-03-08T06:51:18.317+0000] {standard_task_runner.py:60} INFO - Started process 370 to run task
[2024-03-08T06:51:18.360+0000] {standard_task_runner.py:87} INFO - Running: ['airflow', 'tasks', 'run', 'AIRLINES_ETL_PIPELINE', 'runSnowparkCode', 'manual__2024-03-08T06:50:59.288974+00:00', '--job-id', '146', '--raw', '--subdir', 'DAGS_FOLDER/AIRLINES_DATA_ETL_DAG.py', '--cfg-path', '/tmp/tmpie7_itfz']
[2024-03-08T06:51:18.390+0000] {standard_task_runner.py:88} INFO - Job 146: Subtask runSnowparkCode
[2024-03-08T06:51:19.109+0000] {task_command.py:423} INFO - Running <TaskInstance: AIRLINES_ETL_PIPELINE.runSnowparkCode manual__2024-03-08T06:50:59.288974+00:00 [running]> on host f3fe56dfa2ce
[2024-03-08T06:51:19.481+0000] {abstractoperator.py:707} ERROR - Exception rendering Jinja template for task 'runSnowparkCode', field 'sql'. Template: 'MERGE INTO {{ params.database_name }}.{{ params.schema_name }}.OBT_AIRLINES T\n\tUSING {{ params.database_name }}.{{ params.schema_name }}.AIRLINES_STG S\n\t\tON T.PASSENGER_ID = S."passenger_id"\n\t\tAND T.DEPARTURE_DT = TO_DATE(S."Departure_Date")\n\t\tAND T.ARRIVAL_AIRPORT = S."Arrival_Airport"\n\tWHEN MATCHED THEN UPDATE\n\t\tSET \n\t\t\tT.PASSENGER_ID    = S."passenger_id",\n\t\t\tT.FIRST_NAME= S."First_name",\n\t\t\tT.LAST_NAME= S."Last_name",\n\t\t\tT.GENDER= S."Gender",\n\t\t\tT.AGE = S."Age",\n\t\t\tT.NATIONALITY= S."Nationality",\n\t\t\tT.AIRPORT_NAME= S."Airport_name",\n\t\t\tT.COUNTRY_NAME= S."Country_Name",\n\t\t\tT.CONTINENTS= S."Continents",\n\t\t\tT.DEPARTURE_DT = S."Departure_Date",\n\t\t\tT.ARRIVAL_AIRPORT= S."Arrival_Airport",\n\t\t\tT.PILOT_NAME = S."Pilot_Name",\n\t\t\tT.FLIGHT_STATUS = S."Flight_Status"\n\tWHEN NOT MATCHED THEN INSERT \n\t\tVALUES(S."passenger_id",\n\t\t\tS."First_name",\n\t\t\tS."Last_name",\n\t\t\tS."Gender",\n\t\t\tS."Age",\n\t\t\tS."Nationality",\n\t\t\tS."Airport_name",\n\t\t\tS."Country_Name",\n\t\t\tS."Continents",\n\t\t\tS."Departure_Date",\n\t\t\tS."Arrival_Airport",\n\t\t\tS."Pilot_Name",\n\t\t\tS."Flight_Status")\n;\n\t\t\t'
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/abstractoperator.py", line 699, in _do_render_template_fields
    rendered_content = self.render_template(
                       ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/template/templater.py", line 174, in render_template
    return self._render(template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/abstractoperator.py", line 654, in _render
    return super()._render(template, context, dag=dag)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/template/templater.py", line 131, in _render
    return render_template_to_string(template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/helpers.py", line 289, in render_template_to_string
    return render_template(template, cast(MutableMapping[str, Any], context), native=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/helpers.py", line 284, in render_template
    return "".join(nodes)
           ^^^^^^^^^^^^^^
  File "<template>", line 13, in root
  File "/home/airflow/.local/lib/python3.11/site-packages/jinja2/runtime.py", line 852, in _fail_with_undefined_error
    raise self._undefined_exception(self._undefined_message)
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'database_name'
[2024-03-08T06:51:19.500+0000] {taskinstance.py:2728} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 2357, in _run_raw_task
    self._execute_task_with_callbacks(context, test_mode, session=session)
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 2495, in _execute_task_with_callbacks
    task_orig = self.render_templates(context=context)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 2907, in render_templates
    original_task.render_template_fields(context)
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 1241, in render_template_fields
    self._do_render_template_fields(self, self.template_fields, context, jinja_env, set())
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/session.py", line 79, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/abstractoperator.py", line 699, in _do_render_template_fields
    rendered_content = self.render_template(
                       ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/template/templater.py", line 174, in render_template
    return self._render(template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/abstractoperator.py", line 654, in _render
    return super()._render(template, context, dag=dag)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/template/templater.py", line 131, in _render
    return render_template_to_string(template, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/helpers.py", line 289, in render_template_to_string
    return render_template(template, cast(MutableMapping[str, Any], context), native=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/utils/helpers.py", line 284, in render_template
    return "".join(nodes)
           ^^^^^^^^^^^^^^
  File "<template>", line 13, in root
  File "/home/airflow/.local/lib/python3.11/site-packages/jinja2/runtime.py", line 852, in _fail_with_undefined_error
    raise self._undefined_exception(self._undefined_message)
jinja2.exceptions.UndefinedError: 'dict object' has no attribute 'database_name'
[2024-03-08T06:51:19.530+0000] {taskinstance.py:1149} INFO - Marking task as FAILED. dag_id=AIRLINES_ETL_PIPELINE, task_id=runSnowparkCode, execution_date=20240308T065059, start_date=20240308T065118, end_date=20240308T065119
[2024-03-08T06:51:19.583+0000] {logging_mixin.py:188} WARNING - /home/airflow/.local/lib/python3.11/site-packages/airflow/utils/email.py:154 RemovedInAirflow3Warning: Fetching SMTP credentials from configuration variables will be deprecated in a future release. Please set credentials using a connection instead.
[2024-03-08T06:51:19.583+0000] {email.py:270} INFO - Email alerting: attempt 1
[2024-03-08T06:51:21.501+0000] {email.py:281} INFO - Sent an alert email to ['abhijit.bangal92@gmail.com']
[2024-03-08T06:51:23.238+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 146 for task runSnowparkCode ('dict object' has no attribute 'database_name'; 370)
[2024-03-08T06:51:23.284+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-03-08T06:51:23.303+0000] {logging_mixin.py:188} WARNING - /home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py:1201 AirflowProviderDeprecationWarning: Call to deprecated class SnowflakeOperator. (This class is deprecated. Please use `airflow.providers.common.sql.operators.sql.SQLExecuteQueryOperator`. Also, you can provide `hook_params={'warehouse': <warehouse>, 'database': <database>, 'role': <role>, 'schema': <schema>, 'authenticator': <authenticator>,'session_parameters': <session_parameters>}`.)
[2024-03-08T06:51:23.320+0000] {taskinstance.py:3309} INFO - 0 downstream tasks scheduled from follow-on schedule check
